### PR TITLE
Getting curve.cert setup

### PR DESCRIPTION
**Pull request by vsoch**
This includes adding a parameter to the job to ask for creating a "service style DNS entry" which I think is going to be helpful, and cleaning up previous functions (for the stateful set) that we won't need. I did not get to the hosts debugging yet because the broker did not like my curve cert! I was able to figure out setting the correct permission and (likely tomorrow) will debug what it is supposed to look like vs. what I have. Here is how it's honking at me currently:

```bash
kubectl logs -n flux-operator job.batch/flux-sample
Found 6 pods, using pod/flux-sample-0-kglhs
flux-broker: /mnt/curve/curve.cert: invalid CURVE certificate
flux-broker: bootstrap failed
22-09-09 03:09:04 E (zconfig): (1) invalid syntax 'PRIVATE KEY-----'
```
 I also cleaned up the volumes so a minicluster volume is under the same namespace of the job, and the prefixes are set in the same place to hopefully ward against typos. Ping @zekemorton to keep you in the loop! Flagging this as a draft since there is more to do, I'm just getting tired tonight.

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

### CODE REVIEW

**Comment by vsoch** [comment]
okay just pushed a commit https://github.com/flux-framework/flux-operator/pull/15/commits/e8272a945cdca1b02a3053c1a7dc79623e0b0884 that (I think) might handle our curve certificate generation, because now the broker is happy!

```
$ kubectl logs -n flux-operator flux-sample-0-ghfz2
2022-09-09T19:52:12.633864Z broker.err[0]: quorum delayed: waiting for flux-sample-[1-5] (rank 1-5)
2022-09-09T19:53:12.633984Z broker.err[0]: quorum delayed: waiting for flux-sample-[1-5] (rank 1-5)
2022-09-09T19:54:12.634841Z broker.err[0]: quorum delayed: waiting for flux-sample-[1-5] (rank 1-5)
2022-09-09T19:55:12.635502Z broker.err[0]: quorum delayed: waiting for flux-sample-[1-5] (rank 1-5)
```
And waiting for hosts (which we haven't worked on yet, TBA next!) The basic idea is that we know flux is in the user container, so we just use it `flux keygen` (via an InitContainer) to generate the certificate before running our job container.

**Comment by vsoch** [comment]
I want to get the pretty graphic into the README, so since this is working and scoped for a task let's merge (and I'll start a new branch for the networking stuffs).

File: README.md [hunk 1]
```diff
@@ -32,6 +32,12 @@ And you can find the following:
  - [TODO.md](TODO.md) is a set of things to be worked on, if you'd like to contribute!
  - [Documentation](docs) is currently a place to document design, and eventually can be more user-facing docs.
 
+## Design
+
+ - A **MiniCluster** is an [indexed job](https://kubernetes.io/docs/tasks/job/indexed-parallel-processing-static/) so we can create N copies of the "same" base containers (each with flux, and the connected workers in our cluster)
+ - The flux config is written to a volume at `/etc/flux/config` (created via a config map) as a brokers.toml file.
+ - We use an [initContainer](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) with an Empty volume (shared between init and worker) to generate the curve certificates (`/mnt/curve/curve.cert`). The broker sees them via the definition of that path in the broker.toml in our config directory mentioned above.
```
[the remark this answers is not in the document]
Comment (vsoch) [reply README.md:39]: yep - I almost commented that. What is currently happening is technically incorrect, but I think OK for this early development. What is actually happening:

1. there is one initContainer per pod, and since it has a shared volume with it's (TBA) worker pod, it generates a certificate for it. I've confirmed this locally
2. Since the pods aren't actually networked yet, there is no issue

Right now I'm trying to get the core / basics worked out - and the customization of the /etc/hosts is next, and I really just needed the broker to be happy. I suspect once that is working, we will need to:

1. Update the empty volume to generate the certificate somewhere else
2. Have only the first flux worker (do we call it the main one?) generate the curve.cert
3. Create a config map for a volume that is shared between all workers
4. Move it there!

So TLDR we will need to clean up this logic after the entire thing is functioning. For now (development) it's probably OK. I made the comment about curve certificates (plural) because that is the current state (albeit it's wrong and will need to be changed).
